fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/plans/TempTableInsertPlan.java

```diff
@@ -0,0 +1,227 @@
+    }
+
+    private TempTableInsertPlan(@Nonnull final Quantifier.Physical inner,
+                                @Nullable final MessageHelpers.CoercionTrieNode coercionsTrie,
+                                @Nonnull final Value computationValue,
+                                @Nonnull final Value tempTableReferenceValue) {
+        super(inner, null, (Type.Record)((Type.Relation)tempTableReferenceValue.getResultType()).getInnerType(), null,
+                coercionsTrie, computationValue, currentModifiedRecordAlias());
+        this.tempTableReferenceValue = tempTableReferenceValue;
+    }
+
+    @Override
+    public PipelineOperation getPipelineOperation() {
+        return PipelineOperation.INSERT;
+    }
+
+    @Nullable
+    @Override
+    protected <M extends Message> Descriptors.Descriptor getTargetDescriptor(@Nonnull final FDBRecordStoreBase<M> ignored) {
+        return null;
+    }
+
+    @Override
+    public @Nonnull <M extends Message> CompletableFuture<QueryResult> saveRecordAsync(@Nonnull FDBRecordStoreBase<M> store,
+                                                                                       @Nonnull EvaluationContext context,
+                                                                                       @Nonnull M message,
+                                                                                       boolean isDryRun) {
+        // dry run is ignored since inserting into a table queue has no storage side effects.
+        final var queryResult = QueryResult.ofComputed(message);
+        final var tempTable = (TempTable)getTempTableReferenceValue().eval(store, context);
+        tempTable.add(queryResult);
+        return CompletableFuture.completedFuture(queryResult);
+    }
+
+    @Nonnull
+    @Override
+    public TempTableInsertPlan translateCorrelations(@Nonnull final TranslationMap translationMap,
+                                                     @Nonnull final List<? extends Quantifier> translatedQuantifiers) {
+        return new TempTableInsertPlan(
+                Iterables.getOnlyElement(translatedQuantifiers).narrow(Quantifier.Physical.class),
+                getCoercionTrie(),
+                getComputationValue(),
+                getTempTableReferenceValue().translateCorrelations(translationMap));
+    }
+
+    @Nonnull
+    @Override
+    public TempTableInsertPlan withChild(@Nonnull final Reference childRef) {
+        return new TempTableInsertPlan(Quantifier.physical(childRef),
+                getCoercionTrie(),
+                getComputationValue(),
+                getTempTableReferenceValue());
+    }
+
+    @Override
+    public int hashCodeWithoutChildren() {
+        return Objects.hash(BASE_HASH.planHash(PlanHashable.CURRENT_FOR_CONTINUATION), super.hashCodeWithoutChildren());
+    }
+
+    @Override
+    public int planHash(@Nonnull final PlanHashMode mode) {
```

Comment:
You need to hash the `tempTableReferenceValue`

Reply:
done.